### PR TITLE
chore(tests): implements database seeds

### PR DESCRIPTION
**Pull request by hypeJunction**
Database can now be seeded with fake users, groups and objects.

Travis now enables all plugins and seeds the database in memcached and e2e builds.

### CODE REVIEW

**Comment by hypeJunction** [comment]
We can add seeds for other core plugins as the need arises, e.g. when performing migrations.

File: mod/web_services/tests/ElggCoreWebServicesApiTest.php [hunk 1]
```diff
@@ -8,12 +8,24 @@
  */
 class ElggCoreWebServicesApiTest extends ElggCoreUnitTest {
 
+	private $call_method;
+
+	public function setUp() {
+		$this->call_method = get_call_method();
+		if (!$this->call_method) {
+			_elgg_services()->request->server->set('REQUEST_METHOD', 'GET');
+			$this->call_method;
```
Comment (hypeJunction) [inline mod/web_services/tests/ElggCoreWebServicesApiTest.php]: ```$this->call_method = 'GET';```

**Comment by hypeJunction** [comment]
Tests are green. We can improve this a bit once `elgg-cli` is in.

**Review by mrclay — APPROVED** [review]

**Comment by jdalsem** [comment]
does this need to go in 2.3?

**Comment by mrclay** [comment]
Core is committing to supporting 2.3 an awful long time so I think this would be helpful.

**Comment by jdalsem** [comment]
this is adding features... can not go in 2.3... 2.3 is only bugfixes.. sorry

**Comment by mrclay** [comment]
Then alter it so no plugins can register seeds. Hardcode the seed list I guess.

**Comment by hypeJunction** [comment]
This adds another layer of protection to upgrade tests and schema migrations so it makes sense in 2.3.

**Comment by hypeJunction** [comment]
One option to move faker to require-dev and bail if it's not installed, call it a chore until 3.0, then make public

File: engine/classes/Elgg/Database/Seeds/Seed.php [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+	 *
+	 * @param string $msg Message to log
+	 *
+	 * @return void
+	 */
+	public function log($msg, $level = 'NOTICE') {
+
+		if (php_sapi_name() === 'cli') {
+			$handle = $level === 'ERROR' ? STDERR : STDOUT;
+			fwrite(STDOUT, $msg . PHP_EOL);
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/Seeds/Seed.php]: Use $handle

File: engine/classes/Elgg/Database/Seeds/Groups.php [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+				continue;
+			}
+
+			$members_limit = $this->faker->numberBetween(5, 10);
+
+			$members_exclude = [];
+
+			while ($count_members($group) - 1 < $members_limit) {
+
+				$this->log($count_members($group));
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/Seeds/Groups.php]: Remove

File: mod/web_services/tests/ElggCoreWebServicesApiTest.php [hunk 4]
```diff
@@ -8,12 +8,24 @@
  */
 class ElggCoreWebServicesApiTest extends ElggCoreUnitTest {
 
+	private $call_method;
+
+	public function setUp() {
+		$this->call_method = get_call_method();
+		if (!$this->call_method) {
+			_elgg_services()->request->server->set('REQUEST_METHOD', 'GET');
+			$this->call_method = 'GET';
```
Comment (hypeJunction) [inline mod/web_services/tests/ElggCoreWebServicesApiTest.php]: Remove

**Comment by jdalsem** [comment]
> One option to move faker to require-dev and bail if it's not installed, call it a chore until 3.0, then make public

can live with that... but it is probably on the edge of what is allowed in semver

**Comment by hypeJunction** [comment]
This is now a chore

**Comment by jdalsem** [comment]
will you take care of the upmerge?

**Comment by hypeJunction** [comment]
Yes, can do

**Pull request merged**